spec/graphql_response/helpers/response_data_spec.rb

```diff
@@ -0,0 +1,71 @@
+RSpec.describe RSpec::GraphQLResponse, "helper#response", type: :graphql do
+  graphql_query <<-GQL
+    query {
+      characters {
+        id,
+        name,
+        friends {
+          name
+        }
+      }
+    }
+  GQL
+
+  context "has data returned" do
+    it "can return the hash" do
+      expect(response_data).to include(
+        "characters" => [
+          {
+            "id" => "1",
+            "name" => "Jam",
+            "friends" => [
+              { "name" => "Redemption" }
+            ]
+          },
+          {
+            "id" => "2",
+            "name" => "Redemption",
+            "friends" => [
+              { "name" => "Pet" },
+              { "name" => "Jam" }
+            ]
+          },
+          {
+            "id" => "3",
+            "name" => "Pet",
+            "friends" => [
+              { "name" => "Redemption" }
+            ]
+          }
+        ]
+      )
+    end
+
+    it "can dig to a specific layer" do
+      expect(response_data :characters).to include(
+        {
+          "id" => "1",
+          "name" => "Jam",
+          "friends" => [
+            { "name" => "Redemption" }
+          ]
+        },
+        {
+          "id" => "2",
+          "name" => "Redemption",
+          "friends" => [
+            { "name" => "Pet" },
+            { "name" => "Jam" }
+          ]
+        },
+        {
+          "id" => "3",
+          "name" => "Pet",
+          "friends" => [
+            { "name" => "Redemption" }
+          ]
+        }
+      )
+    end
```

Comment:
can you add a spec to show multiple levels of fields?

```ruby
expect(response_data :characters, :friends, :name).to include ...
```

i'd like to have a test around this to make sure i don't accidentally break it in the future. cause honestly, i didn't get what your code was doing until i saw the test and now i'm worried i'll break it 🤣

Reply:
Yeeeeppppppp. Totally agree! Are there any other cases I'm not thinking through? Immediate thoughts are:

1. Test for super, super deep data access.
2. Test to see how errors look when level `x` isn't in the returned response data.

Reply:
those are both good 👍🏻 

for the second one, i kind of think the `response_data` should just return nil as soon as something isn't found. 

and then the `have_operation` matcher can be changed to something like this:

```ruby
it "gets nested stuff" do
  expect(response).to have_data(:foo, :bar, :baz) #=> and if :bar is nil, it raises exception saying "expected to find :foo => :bar => :baz, but :bar was nil
end
```

🤔 

i'll start working on the PR to change the `have_operation` to `have_data`, i think